packages/app/src/app/pages/Sandbox/Editor/Workspace/Chat/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { sortBy, takeRight } from 'lodash-es';
+
+import AutosizeTextArea from '@codesandbox/common/lib/components/AutosizeTextArea';
+import { ENTER } from '@codesandbox/common/lib/utils/keycodes';
+import { useOvermind } from 'app/overmind';
+
+const Container = styled.div`
+  min-height: 200px;
+  max-height: 300px;
+  padding: 0 1rem;
+  color: white;
+  font-size: 0.875rem;
+  display: flex;
+  flex-direction: column;
+  overflow-y: auto;
+`;
+
+const Messages = styled.div`
+  height: 100%;
+  flex: 1;
+`;
+
+export const Chat: React.FC = () => {
+  const [value, setValue] = useState('');
+  const [height, setHeight] = useState('');
+  const { state, actions } = useOvermind();
+  const messagesRef = useRef(null);
+  function scrollDown() {
+    if (messagesRef.current) {
+      messagesRef.current.scrollTop = messagesRef.current.scrollHeight;
+    }
+  }
+  useEffect(scrollDown);
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.keyCode === ENTER && !e.shiftKey) {
+      e.preventDefault();
+      e.stopPropagation();
+      // Enter
+      actions.live.onSendChat({
+        message: value,
+      });
+      setValue('');
+      scrollDown();
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const { messages, users } = state.live.roomInfo.chat;
+  const currentUserId = state.live.liveUserId;
+  const roomInfoUsers = state.live.roomInfo.users;
+
+  return (
+    <Container
+      ref={el => {
+        messagesRef.current = el;
```

Comment:
I wasn't able to test it and refs can be tricky - if someone could take a look, it would be much appreciated! 🙏